Caused by: android.content.res.Resources$NotFoundException: File res/drawable-xxhdpi-v4/*.png from xml type xml resource ID #

I am trying to apply custom icon for the slider but getting following crash. The resource is available in drawable folder.
```
Caused by: android.content.res.Resources$NotFoundException: File res/drawable-xxhdpi-v4/pointing_hand.png from xml type xml resource ID #0x7f0700ab
        at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1243)
        at android.content.res.Resources.loadXmlResourceParser(Resources.java:2172)
        at android.content.res.Resources.getXml(Resources.java:1221)
        at com.ncorti.slidetoact.SlideToActView.parseVectorDrawableCompat(SlideToActView.kt:296)
2019-11-04 13:30:53.979 com.** E/AndroidRuntime:     at com.ncorti.slidetoact.SlideToActView.<init>(SlideToActView.kt:271)
        at com.ncorti.slidetoact.SlideToActView.<init>(SlideToActView.kt:40)
        at com.ncorti.slidetoact.SlideToActView.<init>(Unknown Source:6)
        	... 40 more
     Caused by: java.io.FileNotFoundException: Corrupt XML binary file
        at android.content.res.AssetManager.nativeOpenXmlAsset(Native Method)
        at android.content.res.AssetManager.openXmlBlockAsset(AssetManager.java:1009)
        at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1227)
        	... 46 more
```
## My xml:

```
        <com.ncorti.slidetoact.SlideToActView
            android:id="@+id/btn_swipe"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:layout_below="@+id/home_screen"
            android:layout_margin="30dp"
            app:area_margin="0dp"
            app:elevation="8dp"
            app:slider_icon="@drawable/arrow_right"
            app:inner_color="@color/colorAccent"
            app:outer_color="@color/colorPrimary"
            app:slider_height="64dp"
            app:slider_locked="false"
            app:rotate_icon="true"
            app:text="SWIPE TO REFLECT"
            app:text_color="@android:color/white"
            app:text_size="16sp"
            app:text_style="bold" />
```

## Tesing device:
Samsung A30 - Android version: Pie 9

## Library Version:
```    implementation 'com.ncorti:slidetoact:0.7.0' ```

## Full stacktrace:
```
2019-11-04 13:47:16.887 com.** E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.***, PID: 27796
    android.view.InflateException: Binary XML file line #41: Binary XML file line #41: Error inflating class com.ncorti.slidetoact.SlideToActView
    Caused by: android.view.InflateException: Binary XML file line #41: Error inflating class com.ncorti.slidetoact.SlideToActView
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:686)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:829)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:769)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:902)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:905)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:863)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:554)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:461)
        at com.**.general.HomeFragment.onCreateView(HomeFragment.java:49)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2600)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:881)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:439)
        at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:2079)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1869)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1824)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1727)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2663)
        at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManagerImpl.java:2613)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:246)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:542)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1395)
        at android.app.Activity.performStart(Activity.java:7348)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3147)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1957)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7099)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable-xxhdpi-v4/arrow_right.png from xml type xml resource ID #0x7f07005e
        at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1243)
        at android.content.res.Resources.loadXmlResourceParser(Resources.java:2172)
        at android.content.res.Resources.getXml(Resources.java:1221)
        at com.ncorti.slidetoact.SlideToActView.parseVectorDrawableCompat(SlideToActView.kt:332)
2019-11-04 13:47:16.888 com.** E/AndroidRuntime:     at com.ncorti.slidetoact.SlideToActView.<init>(SlideToActView.kt:306)
        at com.ncorti.slidetoact.SlideToActView.<init>(SlideToActView.kt:39)
        at com.ncorti.slidetoact.SlideToActView.<init>(Unknown Source:6)
        	... 40 more
     Caused by: java.io.FileNotFoundException: Corrupt XML binary file
        at android.content.res.AssetManager.nativeOpenXmlAsset(Native Method)
        at android.content.res.AssetManager.openXmlBlockAsset(AssetManager.java:1009)
        at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1227)
        	... 46 more

```
## Gradle:
```
compileSdkVersion 29
    defaultConfig {
        applicationId "com.***.**"
        minSdkVersion 17
        targetSdkVersion 29
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        vectorDrawables.useSupportLibrary = true
    }
```

Please let me know if you need more details.
Thanks for your support.

--- a/slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt
+++ b/slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt
@@ -7,7 +7,6 @@ import android.animation.ValueAnimator
 import android.annotation.SuppressLint
 import android.content.Context
 import android.content.pm.PackageManager
-import android.content.res.Resources
 import android.content.res.TypedArray
 import android.graphics.Canvas
 import android.graphics.Outline
@@ -31,11 +30,10 @@ import android.view.animation.OvershootInterpolator
 import android.widget.TextView
 import androidx.annotation.RequiresApi
 import androidx.core.content.ContextCompat
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.graphics.drawable.DrawableCompat
 import androidx.core.widget.TextViewCompat
 import androidx.vectordrawable.graphics.drawable.AnimatedVectorDrawableCompat
-import androidx.vectordrawable.graphics.drawable.VectorDrawableCompat
-import org.xmlpull.v1.XmlPullParser
-import org.xmlpull.v1.XmlPullParserException
 
 /**
  *  Class representing the custom view, SlideToActView.
@@ -137,7 +135,7 @@ class SlideToActView @JvmOverloads constructor(
     var iconColor: Int = 0
         set(value) {
             field = value
-            mDrawableArrow.setTint(value)
+            DrawableCompat.setTint(mDrawableArrow, value)
             invalidate()
         }
 
@@ -146,8 +144,10 @@ class SlideToActView @JvmOverloads constructor(
         set(value) {
             field = value
             if (field != 0) {
-                mDrawableArrow = parseVectorDrawableCompat(context.resources, value, context.theme)
-                mDrawableArrow.setTint(iconColor)
+                ResourcesCompat.getDrawable(context.resources, value, context.theme)?.let {
+                    mDrawableArrow = it
+                    DrawableCompat.setTint(it, iconColor)
+                }
                 invalidate()
             }
         }
@@ -201,7 +201,7 @@ class SlideToActView @JvmOverloads constructor(
     private var mTickMargin: Int
 
     /** Arrow drawable */
-    private lateinit var mDrawableArrow: VectorDrawableCompat
+    private lateinit var mDrawableArrow: Drawable
 
     /** Tick drawable, is actually an AnimatedVectorDrawable */
     private val mDrawableTick: Drawable
@@ -404,23 +404,6 @@ class SlideToActView @JvmOverloads constructor(
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             outlineProvider = SlideToActOutlineProvider()
         }
-    }
-
-    private fun parseVectorDrawableCompat(
-        res: Resources,
-        resId: Int,
-        theme: Resources.Theme
-    ): VectorDrawableCompat {
-        val parser = res.getXml(resId)
-        val attrs = Xml.asAttributeSet(parser)
-        var type: Int = parser.next()
-        while (type != XmlPullParser.START_TAG && type != XmlPullParser.END_DOCUMENT) {
-            type = parser.next()
-        }
-        if (type != XmlPullParser.START_TAG) {
-            throw XmlPullParserException("No start tag found")
-        }
-        return VectorDrawableCompat.createFromXmlInner(res, parser, attrs, theme)
     }
 
     override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {

--- a/slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt
+++ b/slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt
@@ -21,7 +21,6 @@ import android.os.Vibrator
 import android.util.AttributeSet
 import android.util.Log
 import android.util.TypedValue
-import android.util.Xml
 import android.view.MotionEvent
 import android.view.View
 import android.view.ViewOutlineProvider

